### PR TITLE
Trims the URL before saving or sending test notification

### PR DESCRIPTION
**Pull request by Abijeet**
Towards https://github.com/TryGhost/Ghost/issues/9263
- trimming the URL value in the validator

Having a space in the URL would cause Ghost to complain - 
> The URL must be in a format like https://hooks.slack.com/services/<your personal key>

Now trimming the URL before validating.

Signed-off-by: Abijeet <abijeetpatro@gmail.com>
- [x] Commit message has a short title & issue references
- [x] Commits are squashed 
- [x] The build will pass (run `ember test` from the repo root - will be `core/client` if working from the submodule in Ghost).

### CODE REVIEW

File: app/validators/slack-integration.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ export default BaseValidator.create({
     properties: ['url'],
 
     url(model) {
-        let url = model.get('url');
+        let url = model.get('url').trim();
```
Comment (kevinansfield) [inline app/validators/slack-integration.js]: It's possible that the url value is `null` or `undefined`, there should be a guard to protect against that when assuming string methods are available, eg:

```javascript
let url = (model.get('url') || '').trim();
```

Comment (kevinansfield) [reply app/validators/slack-integration.js]: or alternatively:

```javascript
let url = model.getWithDefault('url', '').trim();
```

Comment (Abijeet) [reply app/validators/slack-integration.js]:
```
let url = model.getWithDefault('url', '').trim();
```
Good point, thanks, I'll add that.

**Comment by Abijeet** [comment]
@kevinansfield - Thank you. I've made the changes.

**Comment by kevinansfield** [comment]
Hey @Abijeet 👋 Sorry for the slow response, we've been on Christmas break and I'm just getting back into reviewing everything.

I didn't spot this before but validators should not be manipulating data, only validating what's already set on a model.

I think this would be better off being done in the slack app controller's [`updateUrl` action](https://github.com/TryGhost/Ghost-Admin/blob/master/app/controllers/settings/apps/slack.js#L68):

```
        updateURL(value) {
            this.set('model.url', value.trim());
            this.get('model.errors').clear();
        },
```

That will have the effect of not allowing trailing spaces to be entered in the field so the model never gets invalid data set on it.

**Comment by Abijeet** [comment]
@kevinansfield - Thank you. I'll make the changes over the weekend.

**Comment by Abijeet** [comment]
@kevinansfield - Done.

**Pull request merged**